### PR TITLE
feat: lookup flow pages by flowId & stepId

### PR DESCRIPTION
**Pull request by chohner**
following https://github.com/digitalservicebund/a2j-rechtsantragstelle-strapi/pull/293, this PR allows querying for multiple fields and refactors the internal fetch interface to filter by `stepId` and `flowId`

### CODE REVIEW

File: app/services/cms/__test__/getStrapiEntryFromFile.test.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ describe("services/cms", () => {
         expect(
           await getStrapiEntryFromFile({
             apiId: "pages",
-            filterValue: impressumPath,
+            filters: [{ field: "slug", value: impressumPath }],
```
Comment (frederike-ramin) [inline app/services/cms/__test__/getStrapiEntryFromFile.test.ts:67]: Is there also a test for multiple filters?

Comment (frederike-ramin) [reply app/services/cms/__test__/getStrapiEntryFromFile.test.ts:67]: Same for the Api case?

File: app/services/cms/getStrapiEntryFromFile.ts [hunk 2]
```diff
@@ -27,16 +27,29 @@ export const getStrapiEntryFromFile = async ({
...
+        if (nestedFielname === undefined) {
+          return relevantField === value;
+        }
+
+        return (
+          typeof relevantField === "object" &&
+          Array.isArray(relevantField.data) &&
+          relevantField.data.some(
+            (nestedItem) => nestedItem.attributes[nestedFielname] === value,
+          )
```
Comment (frederike-ramin) [inline app/services/cms/getStrapiEntryFromFile.ts:50]: What if the relevant field does not contain an array as `data`? I feel this needs a lot more testing edge cases?

File: app/services/cms/index.server.ts [hunk 3]
```diff
@@ -1,32 +1,28 @@
...
 const getStrapiEntry =
   config().CMS === "FILE" ? getStrapiEntryFromFile : getStrapiEntryFromApi;
 
-export async function fetchMeta(opts: Omit<GetStrapiEntryOpts, "apiId">) {
+export async function fetchMeta(
+  opts: Omit<GetStrapiEntryOpts, "apiId" | "filter"> & { filterValue: string },
+) {
   const populate = "meta";
-  const pageEntry = await getStrapiEntry({ ...opts, apiId: "pages", populate });
+  const filters = [{ value: opts.filterValue, field: "slug" }];
```
Comment (frederike-ramin) [inline app/services/cms/index.server.ts:23]: This still relies on the slug? Do we want to keep it afterall? I thought the plan was to remove it?

File: app/services/cms/index.server.ts [hunk 4]
```diff
@@ -94,7 +95,9 @@ export async function fetchErrors() {
   const cmsErrorSlug = "/error/";
 
   const errorPagePromises = httpErrorCodes.map((errorCode) =>
-    fetchCollectionEntry("pages", `${cmsErrorSlug}${errorCode}`),
+    fetchCollectionEntry("pages", [
+      { field: "slug", value: `${cmsErrorSlug}${errorCode}` },
+    ]),
```
Comment (frederike-ramin) [inline app/services/cms/index.server.ts:100]: For pages, we want to keep the slug? Or just a stepId without a flow to keep it consistent naming-wise? Or maybe that makes it less consistent? wdyt?

Comment (chohner) [reply app/services/cms/index.server.ts:100]: I think I prefer keeping `slug` but up for discussion

File: app/services/cms/getStrapiEntryFromFile.ts (hunk 2 above)
Comment (chohner) [reply app/services/cms/getStrapiEntryFromFile.ts:50]: Then `Array.isArray(relevantField.data)` would fail, right? `data` is standard naming by Strapi, the entry list looks like this:
```typescript
{
  flow_ids: {
    data: [
      {
        attributes: {
          flowId: "foo"
        }
      }
    ]
  }
}
```

File: app/services/cms/__test__/getStrapiEntryFromFile.test.ts (hunk 1 above)
Comment (chohner) [reply app/services/cms/__test__/getStrapiEntryFromFile.test.ts:67]: yes, just not commited 🙈

File: app/services/cms/getStrapiEntryFromFile.ts (hunk 2 above)
Comment (frederike-ramin) [reply app/services/cms/getStrapiEntryFromFile.ts:50]: So that means it always has this structure? What about nested fields that are an array? Or multiply-nested fields for that matter?
I think my problem with this is the function *looks* generic, but it still is very tailored to the two specific cases that we have? And I'm wondering if it would maybe make our lives easier if we just accounted for those two cases instead of building a fit-all filter? Or at the least - add a comment which cases are covered and which aren't?

File: app/services/cms/index.server.ts (hunk 4 above)
Comment (frederike-ramin) [reply app/services/cms/index.server.ts:100]: No need for discussion, okay for me.

File: app/services/cms/getStrapiEntryFromFile.ts (hunk 2 above)
Comment (chohner) [reply app/services/cms/getStrapiEntryFromFile.ts:50]: The app itself sees quite narrow filters, depending on what you want to fetch (`fetchPage(slug: string)` for example).
All those requests then are translated to either a `qs` query string for the REST API or a lookup in the content file (+1 on documenting whats supported).

One alternative to a general `Filters[]` definition would be to introduce two types of filters that we currently support (`property` & `inArray`?) I'll check whether that makes it more or less confusing

Comment (chohner) [reply app/services/cms/getStrapiEntryFromFile.ts]: I'm quite happy with introduction a single optional property, see [`d746597` (#1026)](https://github.com/digitalservicebund/a2j-rechtsantragstelle/pull/1026/commits/d746597365de88a9c74a5e243e7d712deeb1a6fd)
